src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaInstructorTypeService.java

```diff
@@ -1,73 +1,57 @@
 package edu.ucdavis.dss.ipa.services.jpa;
 
-import edu.ucdavis.dss.ipa.entities.InstructorCost;
+import edu.ucdavis.dss.ipa.entities.Instructor;
 import edu.ucdavis.dss.ipa.entities.InstructorType;
+import edu.ucdavis.dss.ipa.entities.Schedule;
+import edu.ucdavis.dss.ipa.entities.UserRole;
 import edu.ucdavis.dss.ipa.repositories.InstructorTypeRepository;
-import edu.ucdavis.dss.ipa.services.InstructorCostService;
 import edu.ucdavis.dss.ipa.services.InstructorTypeService;
+import edu.ucdavis.dss.ipa.services.UserRoleService;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 import javax.inject.Inject;
 import java.util.List;
 
 @Service
 public class JpaInstructorTypeService implements InstructorTypeService {
-    @Inject InstructorTypeRepository instructorTypeRepository;
-    @Inject InstructorCostService instructorCostService;
+    static long INSTRUCTOR_TYPE = 7L;
 
-    @Override
-    public List<InstructorType> findByBudgetId(Long budgetId) {
-        return instructorTypeRepository.findByBudgetId(budgetId);
-    }
+    @Inject InstructorTypeRepository instructorTypeRepository;
+    @Inject UserRoleService userRoleService;
 
     @Override
     public InstructorType findById(Long instructorTypeId) {
         return instructorTypeRepository.findById(instructorTypeId);
     }
 
     @Override
-    @Transactional
-    public void deleteById(long instructorTypeId) {
-        InstructorType instructorType = this.findById(instructorTypeId);
-
-        instructorCostService.removeAssociationByInstructorTypeId(instructorTypeId);
-
-        instructorTypeRepository.deleteById(instructorTypeId);
+    public List<InstructorType> getAllInstructorTypes() {
+        return (List<InstructorType>) instructorTypeRepository.findAll();
```

Comment:
Is the cast necessary?

Reply:
Yes, typically IPA uses List, whereas the repository returns type java.lang.Iterable.